src/Microsoft.ML.Data/Transforms/SelectColumnsTransform.cs

```diff
@@ -0,0 +1,632 @@
+        }
+
+        // Factory method for SignatureLoadDataTransform.
+        public static IDataView Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(ctx, nameof(ctx));
+            SelectColumnsTransform transform;
+
+            // Determine which version of the transform is being loaded.
+            if (CheckModelVersion(ctx, GetDropVersionInfo()))
+            {
+                transform = LoadDropColumnsTransform(env, ctx, input);
+            }
+            else if (CheckModelVersion(ctx, GetChooseVersionInfo()))
+            {
+                transform = LoadChooseColumnsTransform(env, ctx, input);
+            }
+            else
+            {
+                transform = Create(env, ctx);
+            }
+
+            return transform.Transform(input);
+        }
+
+        // Factory method for SignatureDataTransform.
+        private static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            var transform = new SelectColumnsTransform(env, args.KeepHidden, args.Columns);
+            return new SelectColumnsDataTransform(env, transform, new Mapper(transform, input.Schema), input);
+        }
+
+        public void Save(ModelSaveContext ctx)
+        {
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            ctx.Writer.WriteBoolByte(KeepHidden);
+            var length = _selectedColumns.Length;
+            ctx.Writer.Write(length);
+            for (int i = 0; i < length; i++)
+            {
+                ctx.SaveNonEmptyString(_selectedColumns[i]);
+            }
+        }
+
+        public ISchema GetOutputSchema(ISchema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            return new Mapper(this, inputSchema).Schema;
+        }
+
+        public IRowToRowMapper GetRowToRowMapper(ISchema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            return new SelectColumnsDataTransform(_host, this, new Mapper(this, inputSchema), new EmptyDataView(_host, inputSchema));
+        }
+
+        public IDataView Transform(IDataView input)
+        {
+            return new SelectColumnsDataTransform(_host, this, new Mapper(this, input.Schema), input);
+        }
+
+        private sealed class Mapper
+        {
+            private readonly Schema _inputSchema;
+            private readonly Schema _outputSchema;
+            private readonly IHost _host;
+            private readonly SelectColumnsTransform _transform;
+            private readonly int[] _outputToInputMap;
+
+            public ISchema InputSchema => _inputSchema;
+
+            public ISchema Schema => _outputSchema;
+
+            public int this [int colIndex]
```

Comment:
>this [](start = 23, length = 4)

I'm not sure I approve of the use of indexer in this manner. Let's just make a public method